src/substrait/textplan/tests/ParseResultMatchers.h

```diff
@@ -2,29 +2,38 @@
 
 #pragma once
 
+#include <string>
+#include <vector>
+
 #include <gmock/gmock-matchers.h>
 #include <gtest/gtest.h>
 
 #include "substrait/textplan/ParseResult.h"
 
+// NOLINTBEGIN(readability-identifier-naming)
+
 namespace io::substrait::textplan {
```

Comment:
Even if we are using `std::move` semantics I am used to the following rules:

 * If a function will never make a copy of a variable then use `const T&`
 * If a function will always make a copy or might make a copy then use `T`
 * If a function needs to modify a variable then use `T*` (the google style guide suggests `T&` so we could use that too but I find `T*` to be clearer).

Since these functions don't make a copy they should be `const T&` right?  Otherwise we can't do something like...

```
std::vector<std::string> expected_symbols;
if (param_one) {
  expected_symbols = {"x", "y"};
} else {
  expected_symbols = {"a", "b"};
}
for (const auto& test : tests) {
  ASSERT_THAT(test, HasSymbols(expected_symbols));
}
```

We can't use `std::move` and we'd have to make a lot of copies.

Reply:
To be clear, I don't care so much about this particular instance but it would be nice to be aligned on the general rules we want to follow.